### ISSUE TITLE
Adam IO communication Issue

### ISSUE THREAD

**Issue by scienceworld** [museumsvictoria/nodel-recipes#35]
We've had a couple intermittent issues with our AdamIOs recently.  

A Little history
We've added our Exhibit nodes which were updated to support Status to our Dashboard. 

Looking at one of them that i think is running the latest adam io recipe, we've lost communication during a power shutdown and coinidentally the node didnt respond to scheduled enables.

Also with limited Activity history when a supsequent event has come in that's the same type we can't see if there was a command received and it didnt take, or if it didnt receive a command from the exhibit node.
![image](https://user-images.githubusercontent.com/3997916/50921955-19fad280-13fe-11e9-91d4-be2e2fcbb6e3.png)

From the exhibit node I can see that there were two enables, one that didnt turn on the adamio, and one that did.  the first one was via a synched exchange calendar, and the second was done via the exhibit node.
![image](https://user-images.githubusercontent.com/3997916/50922019-47478080-13fe-11e9-9176-7cdb8b9efa6f.png)

So at this point I dont know if there is a failure between teh exhibit node and the adamio node, or a failure between the adamio node and the device.

**Comment by scienceworld** [comment]
I can also add that a second adamio node within the same gallery was also affected.

**Comment by scroix** [comment]
If you want to dig deeper into the logs, try using the API to display and filter all the actions on the activity log.

`http://localhost:8085/REST/nodes/ADAM/logs?max=1000`

![snippet](https://user-images.githubusercontent.com/9277107/50945106-4f2f1f80-14e7-11e9-9f8c-dfa7d8cdc05d.PNG)

**Comment by scienceworld** [comment]
Thanks.  That’s very helpful.  Unfortunately the status updates flooded the log and so I couldn’t go back far enough.  If it does it tomorrow I’ll check again.

**Comment by scienceworld** [comment]
I am able to get the logs.  at 9am today the relay did not close successfully, but a manual interactiion with the exhibit node to enable the exhibit at 9:45 did successfully engage the relay.  

calendar based call to enable the exhibit from the exhibit node:
`08:00","source":"local","type":"event","alias":"MemberOurWorldConveyorBeltRelayPower","arg":"Off"},{"seq":1547654135106,"timestamp":"2019-01-16T09:00:02.220-08:00","source":"remote","type":"event","alias":"Relay3State","arg":false},{"seq":1547654135105,"timestamp":"2019-01-16T09:00:02.173-08:00","source":"local","type":"event","alias":"Power","arg":"On"},{"seq":1547654135104,"timestamp":"2019-01-16T09:00:02.158-08:00","source":"local","type":"event","alias":"MemberOurWorldConveyorBeltRelayPower","arg":"On"},{"seq":1547654135103,"timestamp":"2019-01-16T09:00:02.158-08:00","source":"remote","type":"event","alias":"Relay3State","arg":true},{"seq":1547654135102,"timestamp":"2019-01-16T09:00:00.673-08:00","source":"local","type":"event","alias":"Power","arg":"On"},{"seq":1547654135101,"timestamp":"2019-01-16T09:00:00.673-08:00","source":"local","type":"event","alias":"DesiredPower","arg":"On"},{"seq":1547654135100,"timestamp":"2019-01-16T09:00:00.658-08:00","source":"remote","type":"action","alias":"Relay2On"},{"seq":1547654135099,"timestamp":"2019-01-16T09:00:00.658-08:00","source":"local","type":"action","alias":"Enable"},{"seq":1547654135098,"timestamp":"2019-01-16T09:00:00.658-08:00","source":"remote","type":"event","alias":"Enable"},{"seq":1547654135097,"timestamp":"2019-01-16T09:00:00.658-08:00","source":"local","type":"event","alias":"Power","arg":"Partially {'noPropagation': False, 'state': u'On'}"},{"seq":1547654135096,"timestamp":"2019-01-16T09:00:00.626-08:00","source":"remote","type":"action","alias":"Member OurWorldConveyorBeltRelay Power","arg":{"noPropagation":false,"state":"On"}},{"seq":1547654135095,"timestamp":"2019-01-16T09:00:00.626-08:00","source":"local","type":"event","alias":"DesiredPower","arg":{"noPropagation":false,"state":"On"}},{"seq":1547654135094,"timestamp":"2019-01-16T09:00:00.626-08:00","source":"local","type":"action","alias":"Power","arg":{"noPropagation":false,"state":"On"}},{"seq":1547654135093,"timestamp":"2019-01-16T08:59:12.511-08:00","source":"local","type":"event","alias":"Status","arg":{"level":0,"message":"OK"}},{"seq":1547654135092,"timestamp":"2019-01-16T08:59:12.511-`

manual enable of the exhibit node at 9:45:
`[{"seq":1547654135233,"timestamp":"2019-01-16T09:45:27.758-08:00","source":"local","type":"event","alias":"Status","arg":{"level":0,"message":"OK"}},{"seq":1547654135232,"timestamp":"2019-01-16T09:45:27.758-08:00","source":"local","type":"event","alias":"MemberOurWorldConveyorBeltRelayStatus","arg":{"level":0,"message":"OK"}},{"seq":1547654135231,"timestamp":"2019-01-16T09:45:27.758-08:00","source":"remote","type":"event","alias":"Status","arg":{"level":0,"message":"OK"}},{"seq":1547654135230,"timestamp":"2019-01-16T09:44:22.547-08:00","source":"local","type":"event","alias":"Power","arg":"On"},{"seq":1547654135229,"timestamp":"2019-01-16T09:44:22.547-08:00","source":"local","type":"event","alias":"MemberOurWorldConveyorBeltRelayPower","arg":"On"},{"seq":1547654135228,"timestamp":"2019-01-16T09:44:22.547-08:00","source":"remote","type":"event","alias":"Relay3State","arg":true},{"seq":1547654135227,"timestamp":"2019-01-16T09:44:22.531-08:00","source":"local","type":"event","alias":"Power","arg":"Partially On"},{"seq":1547654135226,"timestamp":"2019-01-16T09:44:22.531-08:00","source":"local","type":"event","alias":"DesiredPower","arg":"On"},{"seq":1547654135225,"timestamp":"2019-01-16T09:44:22.531-08:00","source":"remote","type":"action","alias":"Relay2On"},{"seq":1547654135224,"timestamp":"2019-01-16T09:44:22.531-08:00","source":"local","type":"action","alias":"Enable"},{"seq":1547654135223,"timestamp":"2019-01-16T09:44:12.748-08:00","source":"local","type":"event","alias":"Status","arg":{"level":0,"message":"OK"}},{"seq":1547654135222,"timestamp":"2019-01-16T09:44:12.748-08:00","source":"local","type":"event","alias":"MemberOurWorldConveyorBeltRelayStatus","arg":{"level":0,"message":"OK"}},{"seq":1547654135221,"timestamp":"2019-01-16T09:44:12.748-08:00","source":"remote","type":"event","alias":"Status","arg":{"level":0,"message":"OK"}},{"seq":1547654135220,"timestamp":"2019-01-16T09:42:57.745-08:00","source":"local","type":"event","alias":"Status","arg":{"level":0,"message":"OK"}},{"seq":1547654135219,"timestamp":"2019-01-16T09:42:57.745-08:00","source":"local","type":"event","alias":"MemberOurWorldConveyorBeltRelayStatus","arg":{"level":0,"message":"OK"}},{"seq":1547654135218,"timestamp":"2019-01-16T09:42:57.745-08:00","source":"remote","type":"event","alias":"Status","arg":{"level":0,"message":"OK"}},{"seq":1547654135217,"timestamp":"2019-01-16T09:42:39.619-08:00","source":"local","type":"event","alias":"Power","arg":"Off"},{"seq":1547654135216,"timestamp":"2019-01-16T09:42:39.619-08:00","source":"local","type":"event","alias":"MemberOurWorldConveyorBeltRelayPower","arg":"Off"},{"seq":1547654135215,"timestamp":"2019-01-16T09:42:39.619-08:00","source":"remote","type":"event","alias":"Relay3State","arg":false},{"seq":1547654135214,"timestamp":"2019-01-16T09:42:39.619-08:00","source":"local","type":"event","alias":"Power","arg":"Off"},{"seq":1547654135213,"timestamp":"2019-01-16T09:42:39.619-08:00","source":"local","type":"event","alias":"DesiredPower","arg":"Off"},{"seq":1547654135212,"timestamp":"2019-01-16T09:42:39.619-08:00","source":"remote","type":"action","alias":"Relay2Off"},{"seq":1547654135211,"timestamp":"2019-01-16T09:42:39.603-08:00","source":"local","type":"action","alias":"Disable"},{"seq":1547654135210,"timestamp":"2019-01-16T09:41:42.737-08:00","source":"local","type":"event","alias":"Status","arg":{"level":0,"message":"OK"}},{"seq":1547654135209,"timestamp":"2019-01-16T09:41:42.737-08:00","source":"local","type":"event","alias":"MemberOurWorldConveyorBeltRelayStatus","arg":{"level":0,"message":"OK"}},{"seq":1547654135208,"timestamp":"2019-01-16T09:41:42.737-08:00","source":"remote","type":"event","alias":"Status","arg":`

Adam io relay node receiving 9am calendar event:
`{"seq":1547654128315,"timestamp":"2019-01-16T09:00:27.521-08:00","source":"local","type":"event","alias":"LastContactDetect","arg":"2019-01-16T09:00:27.521-08:00"},{"seq":1547654128314,"timestamp":"2019-01-16T09:00:02.955-08:00","source":"local","type":"event","alias":"Relay2State","arg":false},{"seq":1547654128313,"timestamp":"2019-01-16T09:00:02.955-08:00","source":"local","type":"action","alias":"Relay2State","arg":false},{"seq":1547654128312,"timestamp":"2019-01-16T09:00:01.751-08:00","source":"local","type":"event","alias":"Relay2State","arg":true},{"seq":1547654128311,"timestamp":"2019-01-16T09:00:01.751-08:00","source":"local","type":"action","alias":"Relay2State","arg":true},{"seq":1547654128310,"timestamp":"2019-01-16T09:00:01.751-08:00","source":"local","type":"action","alias":"Relay2Bounce"},{"seq":1547654128309,"timestamp":"2019-01-16T09:00:01.720-08:00","source":"local","type":"event","alias":"Relay3State","arg":false},{"seq":1547654128308,"timestamp":"2019-01-16T09:00:01.720-08:00","source":"local","type":"action","alias":"Relay3State","arg":{"noPropagation":false,"state":"On"}},{"seq":1547654128307,"timestamp":"2019-01-16T09:00:01.611-08:00","source":"local","type":"event","alias":"Relay3State","arg":true},{"seq":1547654128306,"timestamp":"2019-01-16T09:00:01.611-08:00","source":"local","type":"action","alias":"Relay3State","arg":true},{"seq":1547654128305,"timestamp":"2019-01-16T09:00:01.611-08:00","source":"local","type":"action","alias":"Relay3Close"},`

Adam io relay node receiving manual enable from the exhibit node at 9:45:
`[{"seq":1547654128396,"timestamp":"2019-01-16T09:46:42.766-08:00","source":"local","type":"event","alias":"Status","arg":{"level":0,"message":"OK"}},{"seq":1547654128395,"timestamp":"2019-01-16T09:46:42.766-08:00","source":"local","type":"event","alias":"LastContactDetect","arg":"2019-01-16T09:46:42.766-08:00"},{"seq":1547654128394,"timestamp":"2019-01-16T09:45:27.758-08:00","source":"local","type":"event","alias":"Status","arg":{"level":0,"message":"OK"}},{"seq":1547654128393,"timestamp":"2019-01-16T09:45:27.758-08:00","source":"local","type":"event","alias":"LastContactDetect","arg":"2019-01-16T09:45:27.758-08:00"},{"seq":1547654128392,"timestamp":"2019-01-16T09:44:22.547-08:00","source":"local","type":"event","alias":"Relay3State","arg":true},{"seq":1547654128391,"timestamp":"2019-01-16T09:44:22.547-08:00","source":"local","type":"action","alias":"Relay3State","arg":true},{"seq":1547654128390,"timestamp":"2019-01-16T09:44:22.547-08:00","source":"local","type":"action","alias":"Relay3Close"},{"seq":1547654128389,"timestamp":"2019-01-16T09:44:12.748-08:00","source":"local","type":"event","alias":"Status","arg":{"level":0,"message":"OK"}},{"seq":1547654128388,"timestamp":"2019-01-16T09:44:12.748-08:00","source":"local","type":"event","alias":"LastContactDetect","arg":"2019-01-16T09:44:12.748-08:00"},{"seq":1547654128387,"timestamp":"2019-01-16T09:42:57.745-08:00","source":"local","type":"event","alias":"Status","arg":{"level":0,"message":"OK"}},{"seq":1547654128386,"timestamp":"2019-01-16T09:42:57.745-08:00","source":"local","type":"event","alias":"LastContactDetect","arg":"2019-01-16T09:42:57.745-08:00"},{"seq":1547654128385,"timestamp":"2019-01-16T09:42:39.619-08:00","source":"local","type":"event","alias":"Relay3State","arg":false},{"seq":1547654128384,"timestamp":"2019-01-16T09:42:39.619-08:00","source":"local","type":"action","alias":"Relay3State","arg":false},{"seq":1547654128383,"timestamp":"2019-01-16T09:42:39.619-08:00","source":"local","type":"action","alias":"Relay3Open"},{"seq":1547654128382,"timestamp":"2019-01-16T09:41:42.737-08:00","source":"local","type":"event","alias":"Status","arg":{"level":0,"message":"OK"}},`

**Comment by scienceworld** [comment]
This is still a problem.

1) the console isnt logging anything by default
![image](https://user-images.githubusercontent.com/3997916/56169173-26a93c80-5f92-11e9-8109-8da50941e434.png)

2) I can see from the logging api that it seemed to have issued the command to close the relay.  
```{"seq":1552036639096,"timestamp":"2019-04-15T09:15:33.295-07:00","source":"local","type":"event","alias":"LastContactDetect","arg":"2019-04-15T09:15:33.295-07:00"},{"seq":1552036639095,"timestamp":"2019-04-15T09:14:20.671-07:00","source":"local","type":"event","alias":"Relay4State","arg":true},{"seq":1552036639094,"timestamp":"2019-04-15T09:14:20.671-07:00","source":"local","type":"action","alias":"Relay4State","arg":true},{"seq":1552036639093,"timestamp":"2019-04-15T09:14:20.671-07:00","source":"local","type":"action","alias":"Relay4Close"},{"seq":1552036639092,"timestamp":"2019-04-15T09:14:18.295-07:00","source":"local","type":"event","alias":"Status","arg":{"level":0,"message":"OK"}},{"seq":1552036639091,"timestamp":"2019-04-15T09:14:18.295-07:00","source":"local","type":"event","alias":"LastContactDetect","arg":"2019-04-15T09:14:18.295-07:00"},{"seq":1552036639090,"timestamp":"2019-04-15T09:14:08.652-07:00","source":"local","type":"event","alias":"Relay5State","arg":true},{"seq":1552036639089,"timestamp":"2019-04-15T09:14:08.652-07:00","source":"local","type":"action","alias":"Relay5State","arg":true},{"seq":1552036639088,"timestamp":"2019-04-15T09:14:08.652-07:00","source":"local","type":"action","alias":"Relay5Close"},{"seq":1552036639087,"timestamp":"2019-04-15T09:13:03.285-07:00","source":"local","type":"event","alias":"Status","arg":{"level":0,"message":"OK"}},{"seq":1552036639086,"timestamp":"2019-04-15T09:13:03.285-07:00","source":"local","type":"event","alias":"LastContactDetect","arg":"2019-04-15T09:13:03.285-07:00"},{"seq":1552036639085,"timestamp":"2019-04-15T09:11:48.271-07:00","source":"local","type":"event","alias":"Status","arg":{"level":0,"message":"OK"}},{"seq":1552036639084,"timestamp":"2019-04-15T09:11:48.271-07:00","source":"local","type":"event","alias":"LastContactDetect","arg":"2019-04-15T09:11:48.271-07:00"},{"seq":1552036639083,"timestamp":"2019-04-15T09:10:33.264-07:00","source":"local","type":"event","alias":"Status","arg":{"level":0,"message":"OK"}},{"seq":1552036639082,"timestamp":"2019-04-15T09:10:33.264-07:00","source":"local","type":"event","alias":"LastContactDetect","arg":"2019-04-15T09:10:33.264-07:00"},{"seq":1552036639081,"timestamp":"2019-04-15T09:09:18.250-07:00","source":"local","type":"event","alias":"Status","arg":{"level":0,"message":"OK"}},{"seq":1552036639080,"timestamp":"2019-04-15T09:09:18.250-07:00","source":"local","type":"event","alias":"LastContactDetect","arg":"2019-04-15T09:09:18.250-07:00"},{"seq":1552036639079,"timestamp":"2019-04-15T09:08:03.249-07:00","source":"local","type":"event","alias":"Status","arg":{"level":0,"message":"OK"}},{"seq":1552036639078,"timestamp":"2019-04-15T09:08:03.249-07:00","source":"local","type":"event","alias":"LastContactDetect","arg":"2019-04-15T09:08:03.249-07:00"},{"seq":1552036639077,"timestamp":"2019-04-15T09:06:48.243-07:00","source":"local","type":"event","alias":"Status","arg":{"level":0,"message":"OK"}},{"seq":1552036639076,"timestamp":"2019-04-15T09:06:48.243-07:00","source":"local","type":"event","alias":"LastContactDetect","arg":"2019-04-15T09:06:48.243-07:00"},{"seq":1552036639075,"timestamp":"2019-04-15T09:05:33.228-07:00","source":"local","type":"event","alias":"Status","arg":{"level":0,"message":"OK"}},{"seq":1552036639074,"timestamp":"2019-04-15T09:05:33.228-07:00","source":"local","type":"event","alias":"LastContactDetect","arg":"2019-04-15T09:05:33.228-07:00"},{"seq":1552036639073,"timestamp":"2019-04-15T09:04:18.220-07:00","source":"local","type":"event","alias":"Status","arg":{"level":0,"message":"OK"}},{"seq":1552036639072,"timestamp":"2019-04-15T09:04:18.220-07:00","source":"local","type":"event","alias":"LastContactDetect","arg":"2019-04-15T09:04:18.220-07:00"},{"seq":1552036639071,"timestamp":"2019-04-15T09:03:03.219-07:00","source":"local","type":"event","alias":"Status","arg":{"level":0,"message":"OK"}},{"seq":1552036639070,"timestamp":"2019-04-15T09:03:03.219-07:00","source":"local","type":"event","alias":"LastContactDetect","arg":"2019-04-15T09:03:03.219-07:00"},{"seq":1552036639069,"timestamp":"2019-04-15T09:01:48.219-07:00","source":"local","type":"event","alias":"Status","arg":{"level":0,"message":"OK"}},{"seq":1552036639068,"timestamp":"2019-04-15T09:01:48.219-07:00","source":"local","type":"event","alias":"LastContactDetect","arg":"2019-04-15T09:01:48.219-07:00"},{"seq":1552036639067,"timestamp":"2019-04-15T09:00:33.213-07:00","source":"local","type":"event","alias":"Status","arg":{"level":0,"message":"OK"}},{"seq":1552036639066,"timestamp":"2019-04-15T09:00:33.213-07:00","source":"local","type":"event","alias":"LastContactDetect","arg":"2019-04-15T09:00:33.213-07:00"},{"seq":1552036639065,"timestamp":"2019-04-15T09:00:01.345-07:00","source":"local","type":"event","alias":"Relay5State","arg":false},{"seq":1552036639064,"timestamp":"2019-04-15T09:00:01.345-07:00","source":"local","type":"action","alias":"Relay5State","arg":{"noPropagation":false,"state":"On"}},{"seq":1552036639063,"timestamp":"2019-04-15T09:00:01.033-07:00","source":"local","type":"event","alias":"Relay5State","arg":true},{"seq":1552036639062,"timestamp":"2019-04-15T09:00:01.033-07:00","source":"local","type":"action","alias":"Relay5State","arg":true},{"seq":1552036639061,"timestamp":"2019-04-15T09:00:01.033-07:00","source":"local","type":"action","alias":"Relay5Close"},{"seq":1552036639060,"timestamp":"2019-04-15T09:00:01.017-07:00","source":"local","type":"event","alias":"Relay1State","arg":true},{"seq":1552036639059,"timestamp":"2019-04-15T09:00:01.017-07:00","source":"local","type":"action","alias":"Relay1State","arg":true},{"seq":1552036639058,"timestamp":"2019-04-15T09:00:01.017-07:00","source":"local","type":"action","alias":"Relay1Close"},{"seq":1552036639057,"timestamp":"2019-04-15T09:00:00.970-07:00","source":"local","type":"event","alias":"Relay2State","arg":true},{"seq":1552036639056,"timestamp":"2019-04-15T09:00:00.970-07:00","source":"local","type":"action","alias":"Relay2State","arg":true},{"seq":1552036639055,"timestamp":"2019-04-15T09:00:00.970-07:00","source":"local","type":"action","alias":"Relay2Close"},{"seq":1552036639054,"timestamp":"2019-04-15T09:00:00.626-07:00","source":"local","type":"event","alias":"Relay4State","arg":false},{"seq":1552036639053,"timestamp":"2019-04-15T09:00:00.626-07:00","source":"local","type":"action","alias":"Relay4State","arg":{"noPropagation":false,"state":"On"}},{"seq":1552036639052,"timestamp":"2019-04-15T09:00:00.392-07:00","source":"local","type":"event","alias":"Relay3State","arg":true},{"seq":1552036639051,"timestamp":"2019-04-15T09:00:00.392-07:00","source":"local","type":"action","alias":"Relay3State","arg":true},{"seq":1552036639050,"timestamp":"2019-04-15T09:00:00.392-07:00","source":"local","type":"action","alias":"Relay3Close"},{"seq":1552036639049,"timestamp":"2019-04-15T09:00:00.392-07:00","source":"local","type":"event","alias":"Relay4State","arg":true},{"seq":1552036639048,"timestamp":"2019-04-15T09:00:00.392-07:00","source":"local","type":"action","alias":"Relay4State","arg":true},{"seq":1552036639047,"timestamp":"2019-04-15T09:00:00.392-07:00","source":"local","type":"action","alias":"Relay4Close"},{"seq":1552036639046,"timestamp":"2019-04-15T09:00:00.361-07:00","source":"local","type":"event","alias":"Relay2State","arg":true},{"seq":1552036639045,"timestamp":"2019-04-15T09:00:00.353-07:00","source":"local","type":"action","alias":"Relay2State","arg":true},{"seq":1552036639044,"timestamp":"2019-04-15T09:00:00.353-07:00","source":"local","type":"action","alias":"Relay2Close"},{"seq":1552036639043,"timestamp":"2019-04-15T08:59:18.211-07:00","source":"local","type":"event","alias":"Status","arg":{"level":0,"message":"OK"}},{"seq":1552036639042,"timestamp":"2019-04-15T08:59:18.211-07:00","source":"local","type":"event","alias":"LastContactDetect","arg":"2019-04-15T08:59:18.211-07:00"},{"seq":1552036639041,"timestamp":"2019-04-15T08:58:03.210-07:00","source":"local","type":"event","alias":"Status","arg":{"level":0,"message":"OK"}},{"seq":1552036639040,"timestamp":"2019-04-15T08:58:03.210-07:00","source":"local","type":"event","alias":"LastContactDetect","arg":"2019-04-15T08:58:03.210-07:00"},{"seq":1552036639039,"timestamp":"2019-04-15T08:56:48.207-```

3) all the exhibits where we're having this issue are using the mk2 flavour of the adam io recipe.  users are not complaining regularily.

**Comment by justparking** [comment]
The Node activity logs aren't necessary useful. They just tell you when the actions or events are generated. They say little about what going on in the driver.

We need more detailed information from the node itself which is normally dumped on the console.

The available console consists of a relatively small circular buffer so it's hard to capture.

To aid this, I have quickly written a node which should be run on a separate server which adequate disk space to store the entire contents of the console.

You can find it here:
https://gist.github.com/justparking/15a86355b1ce58928523636dd99b0d04

Set the node up and point it to an ADAM that you know shows the problem consistently.

Also remember to enable "debugging" on that ADAM.

**Comment by scroix** [comment]
@paxofsnax Is this related to the issues you were discussing offline with the timeout which would occur at _exactly_ 9:00am each morning.

![dwio03](https://user-images.githubusercontent.com/9277107/56539616-c6c21f80-65a9-11e9-9d0d-5892bad611ec.jpg)

**Comment by scienceworld** [comment]
No, I don’t think so.  Just an intermittent failure of the device or node to send the signal to activate a relay.
only occurs with the MKII adam io recipe.  nodes using the legacy recipe aren’t doing this.
the console doesn’t log anything and I haven’t enabled the debugger yet.

After a firmware upgrade of the devices (hoping that was the issue) the issue occurred again yesterday after 4 or 5 days of working reliably.

J


On Apr 22, 2019, at 4:26 PM, scroix <notifications@github.com<mailto:notifications@github.com>> wrote:


@paxofsnax<https://github.com/paxofsnax> Is this related to the issues you were discussing offline with the timeout which would occur at exactly 9:00am each morning.

[dwio03]<https://user-images.githubusercontent.com/9277107/56539616-c6c21f80-65a9-11e9-9d0d-5892bad611ec.jpg>

**Comment by scienceworld** [comment]
failed again this morning.
From the screenshot we can see from the activity that the relay1 was issued a close action at 9am.  I did a status of relay 1 at 9:15 and it was false.
![image](https://user-images.githubusercontent.com/3997916/56598241-00465b00-65a9-11e9-9f88-dda41f21a922.png)

Whats weird is that I have a relay node that uses all 6 of its relays.  4 of the 6 closed this morning on time and 2 didn't.  It just so happens its the same 2 relays that constitently dont close when the node fails.

I'm wondering if it has something to do with the dashboard, group/members that we have on these nodes.  That is why we upgraded from the legacy to mkII version of the recipe... To attach to the dashboard.

On the adamio with 6 relays, only 2 are configured for group/member/dashboard.  The other 4 are not.  

Could their propagation settings be getting in the way of the relays actually closing?

Here is the exhibit/group node that is triggering the relay that doesnt close reliably.
![image](https://user-images.githubusercontent.com/3997916/56598732-f2dda080-65a9-11e9-8bfb-07b4fd657d4c.png)

Here is the gallery/group node that is triggering the exhibit/group node
![image](https://user-images.githubusercontent.com/3997916/56598808-19034080-65aa-11e9-89b1-4858a06f8006.png)
![image](https://user-images.githubusercontent.com/3997916/56598873-333d1e80-65aa-11e9-9419-f2835abd4e46.png)

Do you think there could be a proegation issue?  something that's setting the value back to false or causing the relay node to miss its command?

**Comment by scienceworld** [comment]
So I actually watched the node today and saw that the console showed all relays turning on except one of them.  Sayed Relay4State False.  When I turned the relay on manually via Adam.Net application the state went to Relay4State True.

![image](https://user-images.githubusercontent.com/3997916/56674963-ddcb4500-666f-11e9-8944-b2df95901e27.png)

I think this is a nodel group/member issue???

the device debug logs are here.

[System - Console Skimmer_nodeconsole_20190424_09.log](https://github.com/museumsvictoria/nodel-recipes/files/3113196/System.-.Console.Skimmer_nodeconsole_20190424_09.log)

**Comment by scienceworld** [comment]
After a chat with @scroix it seems that the adamio recipe does not like getting a relay1close and a relay1state:true at the same time. 
We were calling this node via a exhibit hybrid node with both legacy functions and group action+signal.
When we downgraded the group function to just signal we didnt get a reayxstate noprpegate.

but i'm not entirely certain this adam io recipe supports the group calls.

I tried removing my hybrid exhibit node and just use a group node.  When I used the action for the member of relay1state the adam IO did not close the relay.

so all of our adamio nodes are now continuing to use the exhibit hybrid and we have changed the member call to just signal and no longer with action.
